central/postgres/pruning.go

```diff
@@ -0,0 +1,37 @@
+package postgres
+
+import (
+	"context"
+
+	"github.com/stackrox/rox/central/globaldb"
+	"github.com/stackrox/rox/pkg/logging"
+)
+
+const (
+	pruneActiveComponentStmt = `DELETE FROM active_components child WHERE NOT EXISTS
+		(SELECT * from deployments parent WHERE child.deploymentid = parent.id)`
+
+	pruneClusterHealthStatusStmt = `DELETE FROM cluster_health_statuses child WHERE NOT EXISTS
+		(SELECT * FROM clusters parent WHERE
+		child.Id = parent.Id)`
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+// PruneActiveComponents - prunes active components
+// TODO (ROX-12710):  This will no longer be necessary when the foreign keys are added back
+func PruneActiveComponents(ctx context.Context) {
+	if _, err := globaldb.GetPostgres().Exec(ctx, pruneActiveComponentStmt); err != nil {
+		log.Error(err)
+	}
+}
+
+// PruneClusterHealthStatus - prunes cluster health statuses
+// TODO (ROX-12711):  This will no longer be necessary when the foreign keys are added back
+func PruneClusterHealthStatus(ctx context.Context) {
+	if _, err := globaldb.GetPostgres().Exec(ctx, pruneClusterHealthStatusStmt); err != nil {
+		log.Error(err)
```

Comment:
Add context to what this is err is? The postgres errors can be vague. like "failed to prune cluster health statuses: %v"